src/commands/test/lib/xcuitest-preparer.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from "fs";
+import * as iba from "../../../util/misc/ios-bundle-archiver";
+import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import * as pglob from "../../../util/misc/promisfied-glob";
+import { TestCloudError } from "./test-cloud-error";
+
+export class XCUITestPreparer {
+  private readonly artifactsDir: string;
+  private readonly buildDir: string;
+  private testIpaPath: string;
+
+  constructor(artifactsDir: string, buildDir: string, testIpaPath: string) {
+    if (!artifactsDir) {
+      throw new Error("Argument artifactsDir is required");
+    }
+
+    if (!(buildDir || testIpaPath)) {
+      throw new Error("Either buildDir or testIpaPath argument is required");
+    }
+
+    if (buildDir && testIpaPath) {
```

Comment:
Oh wait, `testIpaPath` refers to the runner?

I've been trying to be strict about referring to the two apps as `runner` and `aut`, because 'testApp' I find quite ambiguous in this case.